src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -1169,10 +1183,47 @@ class module_sram : public module_impl
     }
   }
 
-  [[nodiscard]] const std::vector<std::pair<uint64_t, uint64_t>>&
-  get_ctrlcode_addr_and_size() const override
+  uint32_t*
+  fill_ert_dpu_data(uint32_t *payload) override
   {
-    return m_column_bo_address;
+    auto os_abi = m_parent.get()->get_os_abi();
+    if (os_abi == Elf_Amd_Aie2p) {
```

Comment:
This is me nitpicking, but this function is too long IMO even as it is simple and easy to understand.
Within this class, add two private functions (1) fill_ert_aie2p(payload) and (2) fill_ert_aie2ps(payload).
And I would switch on os_abi().

Unrelated, please change return value of os_abi to plain uint8_t (we don't return a const ref to an integral type).

Reply:
Thanks for all the suggested fixes @stsoe. I addressed all your comments and please take a look again.